src/Ilios/CoreBundle/Controller/ApplicationConfigController.php

```diff
@@ -0,0 +1,321 @@
+<?php
+
+namespace Ilios\CoreBundle\Controller;
+
+use FOS\RestBundle\Controller\Annotations\QueryParam;
+use FOS\RestBundle\Controller\Annotations\RouteResource;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use FOS\RestBundle\Request\ParamFetcherInterface;
+use FOS\RestBundle\Util\Codes;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use FOS\RestBundle\Controller\FOSRestController;
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+use Ilios\CoreBundle\Exception\InvalidFormException;
+use Ilios\CoreBundle\Entity\ApplicationConfigInterface;
+
+/**
+ * Class ApplicationConfigController
+ * @package Ilios\CoreBundle\Controller
+ * @RouteResource("ApplicationConfigs")
+ */
+class ApplicationConfigController extends FOSRestController
+{
+    /**
+     * Get a Application configuration item.
+     *
+     * @ApiDoc(
+     *   section = "Application Configuration",
+     *   description = "Get a application configuration setting.",
+     *   resource = true,
+     *   requirements={
+     *     {
+     *        "name"="id",
+     *        "dataType"="integer",
+     *        "requirement"="\d+",
+     *        "description"="Application configuration setting identifier."
+     *     }
+     *   },
+     *   output="Ilios\CoreBundle\Entity\ApplicationConfig",
+     *   statusCodes={
+     *     200 = "A application configuration setting.",
+     *     404 = "Not Found."
+     *   }
+     * )
+     *
+     * @Rest\View(serializerEnableMaxDepthChecks=true)
+     *
+     * @param $id
+     *
+     * @return Response
+     */
+    public function getAction($id)
+    {
+        $applicationConfig = $this->getOr404($id);
+
+        $authChecker = $this->get('security.authorization_checker');
+        if (! $authChecker->isGranted('view', $applicationConfig)) {
+            throw $this->createAccessDeniedException('Unauthorized access!');
+        }
+
+        $answer['application_configs'][] = $applicationConfig;
```

Comment:
We don't underscore anywhere else.  Also it makes ember-data sad and we have to write custom adapters. Can we use `applicationConfigs` here?